ompi/mpi/c/comm_split_type.c

```diff
@@ -65,6 +65,8 @@ int MPI_Comm_split_type(MPI_Comm comm, int split_type, int key,
         }
 
         if ( MPI_COMM_TYPE_SHARED != split_type && // Same as OMPI_COMM_TYPE_NODE
+             MPI_COMM_TYPE_HW_UNGUIDED != split_type &&
+             MPI_COMM_TYPE_HW_GUIDED != split_type &&
```

Comment:
why not iterating directly over `ompi_comm_split_type_hw_guided_support` ?

Reply:
`ompi_comm_split_type_hw_guided_support` doesn't include `MPI_COMM_TYPE_HW_(UN)GUIDED` or `MPI_UNDEFINED` So it wouldn't be a complete list.

We could make a little helper function to handle the set that is covered (like `bool ompi_comm_is_valid_split_type(int split_type)`) I'm not sure it buys us much though since this is the only place that would be using it.